### PR TITLE
Fix parse email files eml attachment

### PR DESCRIPTION
**Pull request by anara123**
## Status
Ready

## Related Issues
will create an issue soon

## Description
eml contains eml. The inner eml wasn't attached as an attachment in some cases

## Does it break backward compatibility?
   - No

## Must have
- [ ] Tests
- [ ] Documentation (with link to it)
- [ ] Code Review

### CODE REVIEW

**Comment by glicht** [comment]
@anara123  The PR includes the file: Integrations/integration-ESM-v10.yml . This should be in a separate PR.

I've fixed the flake8 errors and put in a small fix so the generated script file is correct.

File: Scripts/ParseEmailFiles/ParseEmailFiles.yml [hunk 1]
```diff
@@ -81,6 +81,6 @@ outputs:
 scripttarget: 0
 runonce: false
 runas: DBotWeakRole
-releaseNotes: Fix - support email files with type 'SMTP Mail'
-tests:
-  - "ParseEmailFiles-test"
```
Comment (glicht) [inline Scripts/ParseEmailFiles/ParseEmailFiles.yml]: Why did you remove the test playbook?

Comment (anara123) [reply Scripts/ParseEmailFiles/ParseEmailFiles.yml]: because the pycharm plugin always reorders my yml, I copy yml from Demisto each time. I forget to add it back. Will add

**Comment by anara123** [comment]
@glicht fixed the build and and your notes

**Comment by anara123** [comment]
@glicht added tests and fixed the build. The failing tests are not related to this PR

**Comment by james-daniel** [comment]
@anara123, to be clear, if the outermost message is of type .msg, and an attached email is of type .eml, this use case is covered? Running into issues with this combination currently.

If I've done my tracing correctly, it seems that when .msg is handled, any .eml files are extracted and saved for additional .eml attachment processing on lines L3252 and L3257 in `Scripts/ParseEmailFiles/ParseEmailFiles.py`, but I don't see a corresponding test case for this scenario. If warranted, I can add a scrubbed sample.

**Comment by anara123** [comment]
@james-daniel should be covered, I didn't add all type of scenarios in the unit tests, but tested manually. But please attach a sample, I will check and if necessary fix

**Comment by james-daniel** [comment]
@anara123 , please see the scrubbed test .msg and .eml here; let me know if there is anything else needed from my end. Thanks

https://github.com/james-daniel/content/commit/4e0bf7295ea4c822d29cbba01a7a320060a554d5

**Comment by anara123** [comment]
@james-daniel the msg file that you provided is not an msg but eml

**Comment by james-daniel** [comment]
@anara123 apologies, for some reason my mail client wrote the msg to disk as eml automagically. Perhaps try this file pair? I had to manually scrub corp info from it.

https://github.com/james-daniel/content/commit/bbf0155a391eb66c6bc5d75e4b8e0042e62c578e

**Comment by james-daniel** [comment]
@anara123 any update on this?

**Comment by anara123** [comment]
@james-daniel no updates yet, will find time next week to work on it

**Comment by james-daniel** [comment]
@anara123 - with the release of [19.3.1](https://github.com/demisto/content/releases/tag/19.3.1) this week, I manually verified that having an .eml embedded within a .msg file is parsed correctly when used in a playbook without having to attach both the .msg _and_ .eml in a given incident. I think it's safe to say this PR covered that scenario as you've mentioned.

This situation occurred when a user-reported phishing email is analyzed by another solution (.msg) which attaches the original email (.eml) as an attachment.

**Comment by anara123** [comment]
Happy to hear that James.
Please feel free to open a ticket or reach me in DFIR if you get any problems with the script.


Sent from my iPhone

On 23 Mar 2019, at 17:35, James Daniel <notifications@github.com<mailto:notifications@github.com>> wrote:


@anara123<https://github.com/anara123> - with the release of 19.3.1<https://github.com/demisto/content/releases/tag/19.3.1> this week, I manually verified that having an .eml embedded within a .msg file is parsed correctly when used in a playbook without having to attach both the .msg and .eml in a given incident. I think it's safe to say this PR covered that scenario as you've mentioned.

This situation occurred when a user-reported phishing email is analyzed by another solution (.msg) which attaches the original email (.eml) as an attachment.
